src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -838,13 +838,41 @@ public void testFileReferenceDuringException(@SysErr Capturable systemErr) {
             invokeMain("-c", getPath("InputMainConfig-classname.xml"),
                     getNonCompilablePath("InputMainIncorrectClass.java"));
         });
-        final String exceptionFirstLine = addEndOfLine("com.puppycrawl.tools.checkstyle.api."
+        final String exceptionMessage = addEndOfLine("com.puppycrawl.tools.checkstyle.api."
                 + "CheckstyleException: Exception was thrown while processing "
                 + new File(getNonCompilablePath("InputMainIncorrectClass.java")).getPath());
-        assertTrue(systemErr.getCapturedData().startsWith(exceptionFirstLine),
+        assertTrue(systemErr.getCapturedData().contains(exceptionMessage),
                 "Unexpected system error log");
     }
 
+    @Test
+    public void testRemoveLexerDefaultErrorListener(@SysErr Capturable systemErr) {
+        assertExitWithStatus(-2, () -> {
+            invokeMain("-t", getNonCompilablePath("InputMainIncorrectClass.java"));
+        });
+
+        assertWithMessage("First line of exception message should not contain lexer error.")
+            .that(systemErr.getCapturedData().startsWith("line 2:2 token recognition error"))
+                .isFalse();
```

Comment:
We remove the default lexer error listener, so that no message appears before the CheckstyleException/ IllegalStateException:

Example:

```bash

line 2:2 token recognition error at: '#'
com.puppycrawl.tools.checkstyle.api.CheckstyleException: IllegalStateException occurred while parsing file /home/nick/Desktop/Tester/src/Test.java.
	at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:104)
	at com.puppycrawl.tools.checkstyle.JavaParser.parseFileText(JavaParser.java:121)
	at com.puppycrawl.tools.checkstyle.JavaParser.parseFile(JavaParser.java:141)
	at com.puppycrawl.tools.checkstyle.AstTreeStringPrinter.printFileAst(AstTreeStringPrinter.java:65)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:285)
	at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:191)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: java.lang.IllegalStateException: 2:0: no viable alternative at input '!'
...
```

vs

```bash
com.puppycrawl.tools.checkstyle.api.CheckstyleException: IllegalStateException occurred while parsing file /home/nick/Desktop/Tester/src/Test.java.
	at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:104)
	at com.puppycrawl.tools.checkstyle.JavaParser.parseFileText(JavaParser.java:121)
	at com.puppycrawl.tools.checkstyle.JavaParser.parseFile(JavaParser.java:141)
	at com.puppycrawl.tools.checkstyle.AstTreeStringPrinter.printFileAst(AstTreeStringPrinter.java:65)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:285)
	at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:191)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: java.lang.IllegalStateException: 2:0: no viable alternative at input '!'
...
```

Reply:
The reason that I wanted to remove the errors that precede the `CheckstyleException` is that there can be several cryptic messages that really don't add any value to bug reports and such.

Reply:
ok, thank you